src/InfrastructureLayer/InfrastructureContainer.cpp

```diff
@@ -14,32 +22,15 @@ InfrastructureContainer::~InfrastructureContainer()
 
 void InfrastructureContainer::setQueueName(const char* queueName)
 {
-    if (settings_->queue() == raceQueue_ ||
-            settings_->queue() == displayQueue_ ||
-            settings_->queue() == debugQueue_ ||
-            settings_->queue() == defaultQueue_)
+    if (settings_->queue() == RACE_QUEUE ||
```

Comment:
Why is this if statement necessary?

Reply:
The default queue names override the custom queue names. Basically, this detects if we already have a custom queue name in place.

For example, without this statement:
If we put in a custom queue name `customQueue` into the config file, and try to run it in Race mode, The `InfrastructureContainer` is created by reading in the config file and the queue name at this point will be `customQueue`. However, the `EpsilonDashboard` then parses the input arguments recognizes that it is running in Race mode and therefore resets the queue name to be `raceQueue`.